.github/workflows/build_and_test.yml

```diff
@@ -410,7 +411,7 @@ jobs:
     name: Build and test pytket (macos-arm64)
     runs-on: ['self-hosted', 'macOS', 'ARM64']
     needs: check_changes
-    if: github.event_name == 'push' || github.event_name == 'schedule' || github.event.pull_request.head.repo.full_name == github.repository
+    if: github.event_name == 'push' || github.event_name == 'schedule' || github.event.pull_request.head.repo.full_name == github.repository || github.event.workflow_dispatch.repository == github.repository
```

Comment:
This is possible according to the github reference, but not sure if it's necessary. Maybe `github.event_name == 'workflow_dispatch'` would be enough the workflow dispatch case? Not 100% why it's necessary for the 'pull_request" case...

Reply:
It's because we don't want PRs from forks to run on the self-hosted Mac M1 (a bit of paranoia).